### PR TITLE
enhacing delete from array by name, previously was only by index

### PR DESCRIPTION
**Pull request by qu1queee**
Can we add this enhancement, so we can delete an element of array based on his name, rather than the index. We need more use cases for arrays nested in arrays, but so far for the current use cases in the test files, this should work.
Regards,
Enrique

### CODE REVIEW

File: delete_test.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ mapArray:
   boo: laa`
 
 		yml, err = simpleyaml.NewYaml([]byte(yaml))
-		Expect(err).NotTo(HaveOccurred())
+		Expect(err).NotTo(	HaveOccurred())
```
Comment (MissingRoberto) [inline delete_test.go:35]: can you get rid of this extra tabulation?

File: delete_test.go [hunk 2]
```diff
@@ -43,7 +43,12 @@ mapArray:
 		Expect(err).To(MatchError("property not found"))
 	})
 
-	It("should delete a value from an array ", func() {
+	It("should delete a value from an array based on name", func() {
+		err = Delete(yml, "array.bar")
+		Expect(err).NotTo(HaveOccurred())
```
Comment (MissingRoberto) [inline delete_test.go:48]: can you check if it was correctly deleted here?

File: delete.go [hunk 3]
```diff
@@ -37,25 +37,49 @@ func DeleteInFile(file string, path string) error {
...
+}
+
 func Delete(yml *simpleyaml.Yaml, path string) error {
 	propsArr := strings.Split(path, ".")
 	propName := propsArr[len(propsArr)-1]
 	props := propsArr[:len(propsArr)-1]
 	newPath := strings.Join(props, ".")
 
-	if index, err := strconv.Atoi(propName); err == nil {
+	yaml_pointer, _ := get(yml, newPath)
```
Comment (MissingRoberto) [inline delete.go:56]: required error handling

File: delete.go [hunk 4]
```diff
@@ -37,25 +37,49 @@ func DeleteInFile(file string, path string) error {
...
+	path_array, _ := yaml_pointer.Array()
```
Comment (MissingRoberto) [inline delete.go:57]: required error handling

**Comment by herrjulz** [comment]
hi @qu1queee thank you for you PR. 
Unfortunately, i need to tell you that this is already possible with `goml`. You just need to use the right syntax to do so. I have examples in the [README](https://github.com/JulzDiverse/goml#examples-2). 

Basically, if you want to delete an entry of an _array_ with _maps_ you can specify any identifier to delete this entry. For example:

Consider the following YAML...

```
---
someArray:
- name: foo
- name: bar

stringArray:
- foo
- bar
- foobar
```

a) If you want to delete "foo" from "someArray", you can do so with the following `goml` call:

```
$ goml delete -f file.yml -p someArray.name:foo 
```

b) If you want do delete an entry of array of strings, it is also possible. Lets say you want to delete "bar" from "stringArray":

 ```
$ goml delete -f file.yml -p stringArray.:bar
```

Here is the related code line: [goml delete](https://github.com/JulzDiverse/goml/blob/master/delete.go#L59)

**Comment by qu1queee** [comment]
@JulzDiverse I though the `:` was used for hashes, as a way to specify the key identifier. I do not see why would you use an `:` when dealing with simple arrays, its a different structure. Can you explain more in detail your `b)` example, i´m confused.

File: delete_test.go (hunk 1 above)
Comment (qu1queee) [reply delete_test.go:35]: sure, thanks.

File: delete_test.go (hunk 2 above)
Comment (qu1queee) [reply delete_test.go:48]: sure, thanks.

File: delete.go (hunk 4 above)
Comment (qu1queee) [reply delete.go:57]: sure, thanks.

**Comment by herrjulz** [comment]
@qu1queee in the `goml` syntax you use `:` as key-value separator to identify values inside a map or array. As an array with values only does not contain a key, you simply leave it empty. That's why you do it like this `some.path.to.array.:value`. It is simply a separation syntax: 

- For index you use a number 
- for strings you specify a `<key>:<value>` and if there is no key you leave it empty :) 
 
Examples are [HERE](https://github.com/JulzDiverse/goml#examples-1)

One more note to that:

The same syntax is used to identify an array entry with set (to update a property). Consider you want to add a new entry to the array. You would simply specify the following path `path.to.array` and specify the value you want to add. But what if you want to update a specific value in an array without adding a duplicate entry? The `goml` syntax to do so is: `path.to.arry.:value`. The same syntax is used for `delete`. I wanted to keep the syntax consistent.

**Pull request closed without merging**